src/content/1.7/modules/sample_modules/view-order-hooks-demo.md

```diff
@@ -0,0 +1,522 @@
+        $installer = InstallerFactory::create();
+
+        return $installer->uninstall() && parent::uninstall();
+    }
+
+}
+```
+
+Let's create `Signature` entity class:
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table()
+ * @ORM\Entity()
+ */
+class Signature
+{
+    /**
+     * @var int|null
+     *
+     * @ORM\Id
+     * @ORM\Column(name="id_signature", type="integer")
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var int
+     *
+     * @ORM\Column(name="id_order", type="integer")
+     */
+    private $orderId;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(name="filename", type="string")
+     */
+    private $filename;
+
+    /**
+     * @return int|null
+     */
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return Signature
+     */
+    public function setId(int $id): Signature
+    {
+        $this->id = $id;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFilename(): string
+    {
+        return $this->filename;
+    }
+
+    /**
+     * @param string $filename
+     *
+     * @return Signature
+     */
+    public function setFilename(string $filename): Signature
+    {
+        $this->filename = $filename;
+
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getOrderId(): int
+    {
+        return $this->orderId;
+    }
+
+    /**
+     * @param int $orderId
+     *
+     * @return Signature
+     */
+    public function setOrderId(int $orderId): Signature
+    {
+        $this->orderId = $orderId;
+
+        return $this;
+    }
+}
+
+```
+
+Lets create `SignatureRepository` class:
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Repository;
+
+use Doctrine\ORM\EntityRepository;
+
+class SignatureRepository extends EntityRepository
+{
+}
+```
+
+Lets create services configuration for the above classes to use dependency injection (DI) in `config/services.yml`:
+
+```yaml
+parameters:
+  signatureImgDirectory: 'signatures/'
+
+services:
+  prestashop.module.demovieworderhooks:
+    class: DemoViewOrderHooks
+    factory: [Module, getInstanceByName]
+    arguments:
+      - 'demovieworderhooks'
+
+  prestashop.module.demovieworderhooks.repository.order_repository:
+    class: PrestaShop\Module\DemoViewOrderHooks\Repository\OrderRepository
+
+  prestashop.module.demovieworderhooks.repository.signature_repository:
+    class: PrestaShop\Module\DemoViewOrderHooks\Repository\SignatureRepository
+    factory: ['@doctrine.orm.default_entity_manager', getRepository]
+    arguments:
+      - PrestaShop\Module\DemoViewOrderHooks\Entity\Signature
+```
+
+Let's create `SignaturePresenter` class responsible for returning order customer data:
+
+```php
+<?php
+
```

Comment:
Previuosly declaring strict types just after php tag and now here is a space between them. Be consistent